src/garage/tf/models/gaussian_mlp_model2.py

```diff
@@ -0,0 +1,248 @@
+                 std_hidden_nonlinearity=tf.nn.tanh,
+                 std_hidden_w_init=tf.initializers.glorot_uniform(),
+                 std_hidden_b_init=tf.zeros_initializer(),
+                 std_output_nonlinearity=None,
+                 std_output_w_init=tf.initializers.glorot_uniform(),
+                 std_parameterization='exp',
+                 layer_normalization=False):
+        # Network parameters
+        super().__init__(name)
+        self._hidden_sizes = hidden_sizes
+        self._output_dim = output_dim
+        self._learn_std = learn_std
+        self._adaptive_std = adaptive_std
+        self._std_share_network = std_share_network
+        self._std_hidden_sizes = std_hidden_sizes
+        self._min_std = min_std
+        self._max_std = max_std
+        self._std_hidden_nonlinearity = std_hidden_nonlinearity
+        self._std_hidden_w_init = std_hidden_w_init
+        self._std_hidden_b_init = std_hidden_b_init
+        self._std_output_nonlinearity = std_output_nonlinearity
+        self._std_output_w_init = std_output_w_init
+        self._std_parameterization = std_parameterization
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._hidden_w_init = hidden_w_init
+        self._hidden_b_init = hidden_b_init
+        self._output_nonlinearity = output_nonlinearity
+        self._output_w_init = output_w_init
+        self._output_b_init = output_b_init
+        self._layer_normalization = layer_normalization
+
+        # Tranform std arguments to parameterized space
+        self._init_std_param = None
+        self._min_std_param = None
+        self._max_std_param = None
+        if self._std_parameterization == 'exp':
+            self._init_std_param = np.log(init_std)
+            if min_std is not None:
+                self._min_std_param = np.log(min_std)
+            if max_std is not None:
+                self._max_std_param = np.log(max_std)
+        elif self._std_parameterization == 'softplus':
+            self._init_std_param = np.log(np.exp(init_std) - 1)
+            if min_std is not None:
+                self._min_std_param = np.log(np.exp(min_std) - 1)
+            if max_std is not None:
+                self._max_std_param = np.log(np.exp(max_std) - 1)
+        else:
+            raise ValueError("std parameterization should be or 'exp' or "
+                             "'softplus' but got {}".format(
+                                 self._std_parameterization))
+
+    # pylint: disable=arguments-differ
+    def _build(self, state_input, name=None):
+        """Build model.
+
+        Args:
+            state_input (tf.Tensor): Entire time-series observation input.
+            name (str): Inner model name, also the variable scope of the
+                inner model, if exist. One example is
+                garage.tf.models.Sequential.
+
+        Returns:
+            tfp.distributions.MultivariateNormalDiag: Distribution.
+
+        """
+        del name
```

Comment:
strange that it doesn't use name? does the tensorboard look correct?

Reply:
This name is for the inner model, if there's one, e.g. there will be multiple models in `garage.tf.models.Sequential` so each one would need a unique name to distinguish. In other cases, model will always have the same name, i.e. its own name, while the policy that uses it (i.e. the policy which calls `build()` will have different name, so we will have `policy1/MLPModel` and `policy2/MLPModel`.